File: DigitalLibrary.Models/AccountingModels/AccountingDto.cs
using System;
using System.ComponentModel.DataAnnotations;
using DigitalLibrary.Models.BookModels;
using DigitalLibrary.Models.Entities;
using DigitalLibrary.Models.Enums;
using DigitalLibrary.Models.LibraryModels;
using DigitalLibrary.Models.ProfileModels;
using DigitalLibrary.Models.StorageModels;

namespace DigitalLibrary.Models.AccountingModels
{
    public class AccountingDto
    {
        public Guid Id { get; set; }

        [Required]
        public Status Status { get; set; }

        [Required]
        public DateTime RequestDate { get; set; }

        public DateTime IssueDate { get; set; }

        [Required]
        public DateTime ReturnDate { get; set; }

        public DateTime DateReturned { get; set; }

        public LibraryDto Library { get; set; }
        public BookDto Book { get; set; }
        public StoredItemDto Storage { get; set; }
        public ProfileDto Profile { get; set; }
    }
}

File: DigitalLibrary.Client/NGClient/src/app/shared/entities/accounting.dto.ts
import {Library} from "./library.entity";
import {Book} from "./book.entity";
import {StoredItem} from "./storedItem.entity";
import { Profile } from "../../services/profile.service";

export class accountingDto {
  id: string;
  status: string;
  requestDate: Date;
  issueDate: Date;
  returnDate: Date;
  dateReturned: Date;
  library: Library;
  book: Book;
  storage: StoredItem;
  profile: Profile;
}
